ie_serving/models/gs_model.py

```diff
@@ -123,13 +124,14 @@ def get_engine_for_version(cls, model_name, version_attributes):
         logger.info('Downloaded files from GCS')
 
         engine_spec = cls._get_engine_spec(model_name, version_attributes)
-        engine = IrEngine.build(**engine_spec)
+        engine_process = multiprocessing.Process(target=IrEngine.build,
+                                                 args=engine_spec)
 
```

Comment:
The engine process must start somewhere. No matter where you decide to call `start()` on this handler you can't call `cls.delete_local_mirror(...)` like you did below. It would cause files deletion before engine process can properly load them. 
Consider moving files download and deletion to engine process.